src/transaction/buffer.rs

```diff
@@ -34,24 +33,24 @@ impl InnerBuffer {
 
 /// A caching layer which buffers reads and writes in a transaction.
 pub struct Buffer {
-    mutations: Mutex<InnerBuffer>,
+    inner: InnerBuffer,
 }
 
 impl Buffer {
     pub fn new(is_pessimistic: bool) -> Buffer {
         Buffer {
-            mutations: Mutex::new(InnerBuffer::new(is_pessimistic)),
+            inner: InnerBuffer::new(is_pessimistic),
```

Comment:
If we don't have the Mutex, then I don't think we need the inner/outer split.

Reply:
Yes. I hoped to make the PR simple :). Will update it.